File: VB/DXApplication5/MVVM/Views/CustomersView.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports DXApplication5.MVVM.ViewModels
Imports DevExpress.XtraBars.Docking2010
Imports DXApplication5.MVVM.Data
Imports DevExpress.XtraGrid.Views.Base

Namespace DXApplication5.MVVM.Views
	Partial Public Class CustomersView
		Inherits DevExpress.XtraEditors.XtraUserControl
		Public Sub New()
			InitializeComponent()
			If (Not mvvmContext.IsDesignMode) Then
				InitializeBindings()
			End If
		End Sub
		Private Sub InitializeBindings()
			mvvmContext.ViewModelType = GetType(CustomersViewModel)
			Dim fluent = mvvmContext.OfType(Of CustomersViewModel)()
			fluent.SetBinding(customersGridControl, Function(c) c.DataSource, Function(x) x.Entities)
            fluent.BindCommand(TryCast(windowsUIButtonPanel.Buttons(0), WindowsUIButton), Sub(x) x.FindCustomer())
            fluent.WithEvent(Of ColumnView, FocusedRowObjectChangedEventArgs)(gridView1, "FocusedRowObjectChanged").SetBinding(Function(x) x.SelectedEntity, _
                                           Function(args) TryCast(args.Row, Customer), Sub(gView, entity) gView.FocusedRowHandle = gView.FindRow(entity))

		End Sub
	End Class
End Namespace

File: CS/DXApplication5/MVVM/Views/CustomersView.cs
using System;
using System.Collections.Generic;
using System.Linq;
using DXApplication5.MVVM.ViewModels;
using DevExpress.XtraBars.Docking2010;
using DXApplication5.MVVM.Data;
using DevExpress.XtraEditors;
using DevExpress.Utils.MVVM.Services;

namespace DXApplication5.MVVM.Views {
    public partial class CustomersView : DevExpress.XtraEditors.XtraUserControl {
        public CustomersView() {
            InitializeComponent();
            if (!mvvmContext.IsDesignMode) {
                InitializeBindings();
            }
        }
        private void InitializeBindings() {
            mvvmContext.ViewModelType = typeof(CustomersViewModel);
            var fluent = mvvmContext.OfType<CustomersViewModel>();
            fluent.SetBinding(customersGridControl, c => c.DataSource, x => x.Entities);
            fluent.BindCommand(windowsUIButtonPanel.Buttons[0] as WindowsUIButton, x => x.FindCustomer());
            fluent.WithEvent<DevExpress.XtraGrid.Views.Base.ColumnView, DevExpress.XtraGrid.Views.Base.FocusedRowObjectChangedEventArgs>(gridView1, "FocusedRowObjectChanged")
                .SetBinding(x => x.SelectedEntity,
                    args => args.Row as Customer,
                    (gView, entity) => gView.FocusedRowHandle = gView.FindRow(entity));
            
        }
    }
}
